site/src/lib/course-planner/CourseSearch.ts

```diff
@@ -72,6 +73,40 @@ export function searchCourses(input: string, courseLookup:
                 }
             }
         }
+    } 
+    
+    // If the search input is just numbers, match courses with that number
+    const deptList = ["AASP", "AAST", "ABRM", "AGNR", "AGST", "AMSC", "AMST", "ANSC", "ANTH", "AOSC", "ARAB", "ARCH", "AREC", "ARHU", "ARMY", "ARSC", "ARTH", "ARTT", "ASTR", "BCHM", "BIOE", "BIOI", "BIOL", "BIOM", "BIPH", "BISI", "BMGT", "BMSO", "BSCI", "BSOS", "BSST", "BUAC", "BUDT", "BUFN", "BULM", "BUMK", "BUSI", "BUSM", "BUSO", "CBMG", "CCJS", "CHBE", "CHEM", "CHIN", "CHPH", "CHSE", "CINE", "CLAS", "CLFS", "CMLT", "CMSC", "COMM", "CPBE", "CPCV", "CPET", "CPGH", "CPJT", "CPMS", "CPPL", "CPSA", "CPSF", "CPSG", "CPSN", "CPSP", "CPSS", "DANC", "DATA", "ECON", "EDCP", "EDHD", "EDHI", "EDMS", "EDSP", "EDUC", "EMBA", "ENAE", "ENBC", "ENCE", "ENCO", "ENEB", "ENEE", "ENES", "ENFP", "ENGL", "ENMA", "ENME", "ENPM", "ENRE", "ENSE", "ENSP", "ENST", "ENTM", "ENTS", "EPIB", "FGSM", "FIRE", "FMSC", "FREN", "GBHL", "GEMS", "GEOG", "GEOL", "GERS", "GREK", "GVPT", "HACS", "HBUS", "HDCC", "HEBR", "HESI", "HESP", "HGLO", "HHUM", "HISP", "HIST", "HLSA", "HLSC", "HLTH", "HNUH", "HONR", "IDEA", "IMDM", "IMMR", "INAG", "INFM", "INST", "ISRL", "ITAL", "JAPN", "JOUR", "JWST", "KNES", "KORA", "LACS", "LARC", "LATN", "LBSC", "LEAD", "LGBT", "LING", "MAIT", "MATH", "MEES", "MIEH", "MITH", "MLAW", "MSML", "MSQC", "MUED", "MUSC", "NACS", "NAVY", "NEUR", "NFSC", "NIAP", "NIAV", "PEER", "PERS", "PHIL", "PHPE", "PHSC", "PHYS", "PLCY", "PLSC", "PORT", "PSYC", "RDEV", "RELS", "RUSS", "SLAA", "SLLC", "SMLP", "SOCY", "SPAN", "SPHL", "STAT", "SURV", "TDPS", "THET", "TLPL", "TLTC", "UMEI", "UNIV", "URSP", "USLT", "VMSC", "WEID", "WGSS", "XPER"];
+    if (simpleInput.length >= 1 && /^[0-9]+$/i.test(simpleInput)) {
+
+        // get all the courses from every department 
+        const allDeptCourses: Record<string, Course> = {};
+        for (const dept of deptList) {
+            const deptCourses = courseLookup[dept];
+            if (deptCourses !== undefined) {
+                for (const courseCode in deptCourses) {
+                    const uniqueCourseCode = `${dept}-${courseCode}`;
+                    allDeptCourses[uniqueCourseCode] = deptCourses[courseCode];
+                }
+            }
+        }
+
+        for (const courseCode in allDeptCourses) {
+            let shouldBeInResult = true;
+            if (simpleInput.length > courseCode.length) {
+                shouldBeInResult = false;
+            } else {
+                const courseNumber = courseCode.substring(5);
+                for (let i = 0; i < simpleInput.length; i++) { 
+                    if (simpleInput[i] != courseNumber[i]) {
+                        shouldBeInResult = false;
+                    }
+                }
+            }
+            if (shouldBeInResult) {
+                result.push(allDeptCourses[courseCode]);
+            }
+        }
```

Comment:
Suggestion: If you decide to go with the route (see my prev. comment) of only accepting number inputs that are three numbers long (ex. "417" and not "4" or "41"), then you won't need to iterate through all courses since you'll be able to just do lookups like `courseLookup["CMSC"]["417"]` (I think) and can just do that once for each department code. If you don't decide to do that though then this lgtm.